### PR TITLE
Rem 217

### PR DESCRIPTION
**Pull request by NoraBitarSM**

### CODE REVIEW

**Comment by NoraBitarSM** [comment]
functionality tested, approved

File: src/components/Shared/AddressForm.js [hunk 1]
```diff
@@ -5,57 +5,47 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
...
-
+  function onAddressSubmit(post) {
+    const data = { ...post, recordId: recordId }
+    postRequest({
+      extension: SystemRepository.Address.set,
+      record: JSON.stringify(data)
+    }).then(res => {
+      data.addressId = res.recordId
+      onSubmit(data)
+      if (res.recordId) toast.success('Record Updated Successfully')
```
Comment (RebeccaFrem02) [inline src/components/Shared/AddressForm.js:21]: Make sure that if(res.recordId) is required.

File: src/components/Shared/AddressForm.js [hunk 2]
```diff
@@ -5,57 +5,47 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
 import { AddressFormShell } from 'src/components/Shared/AddressFormShell'
```
Comment (RebeccaFrem02) [inline src/components/Shared/AddressForm.js:5]: The comments in line 1 and 3 serve no purpose.

File: src/components/Shared/AddressForm.js [hunk 3]
```diff
@@ -5,57 +5,47 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
...
-return (
-  <>
-  <AddressFormShell
-     editMode={editMode}
-     setAddress={setAddress}
-     address={address}
-     allowPost={true}
-     onSubmit={onAddressSubmit}
-    />
+        .catch(error => {})
```
Comment (RebeccaFrem02) [inline src/components/Shared/AddressForm.js:36]: Remove.

File: src/components/Shared/AddressForm.js (hunk 1 above)
Comment (Ghinwa-Arab) [reply src/components/Shared/AddressForm.js:21]: yes needed

**Review by Alinadimmokdad — COMMENTED** [review]
Ali's review

File: src/components/Shared/AddressForm.js [hunk 4]
```diff
@@ -17,22 +15,20 @@ const AddressForm = ({ recordId, address, setAddress, editMode, onSubmit }) => {
...
-      var parameters = `_filter=` + '&_recordId=' + recordId
-      getRequest({
-        extension: SystemRepository.Address.get,
-        parameters: parameters
-      })
-        .then(res => {
-          var result = res.record
-          setAddress(result)
+    ;(async function () {
+      setAddress([])
```
Comment (hadichahine) [inline src/components/Shared/AddressForm.js:23]: why empty then fetching?

File: src/pages/address-book/index.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useInvalidate, useResourceQuery } from 'src/hooks/resource'
+import { useWindow } from 'src/windows'
+import AddressForm from 'src/components/Shared/AddressForm'
+
+const AddressBook = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const [address, setAddress] = useState([])
```
Comment (hadichahine) [inline src/pages/address-book/index.js:15]: Why do we want this here at all?

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**